http/legacy/write_handler.go

```diff
@@ -155,7 +156,15 @@ func (h *WriteHandler) findBucket(ctx context.Context, orgID influxdb.ID, db, rp
 		return nil, err
 	}
 
-	return h.BucketService.FindBucketByID(ctx, mapping.BucketID)
+	bucket, err := h.BucketService.FindBucketByID(ctx, mapping.BucketID)
+	if err != nil {
+		return nil, err
+	}
+	_, _, err = authorizer.AuthorizeWrite(ctx, influxdb.BucketsResourceType, bucket.ID, bucket.OrgID)
```

Comment:
This is the only hard-coded auth check at this layer, all others are abstracted away behind various middlewares. Not sure if/how it could be refactored to better fit existing patterns.

Reply:
This is great and does follow the same pattern as `/api/v2/write`:

https://github.com/influxdata/influxdb/blob/fcb70c26d54b78d8da1bab9300273ae311f6c600/http/write_handler.go#L174-L177

Can you make the changes so the code looks the same – i.e. move this check outside the `findBucket` function into the `handleWrite` function after the `findBucket` call?